src/mw-modal/services/mw_modal.js

```diff
@@ -110,30 +133,53 @@ angular.module('mwUI.Modal')
        * @description Shows the modal
        */
       this.show = function () {
+        var dfd = $q.defer();
```

Comment:
does this https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns#the-deferred-anti-pattern apply here?